python/samples/getting_started_with_agents/step4_kernel_function_strategies.py

```diff
@@ -0,0 +1,118 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+import asyncio
+
+from semantic_kernel.agents import AgentGroupChat, ChatCompletionAgent
+from semantic_kernel.agents.strategies import (
+    KernelFunctionSelectionStrategy,
+    KernelFunctionTerminationStrategy,
+)
+from semantic_kernel.connectors.ai.open_ai.services.azure_chat_completion import AzureChatCompletion
+from semantic_kernel.contents.chat_message_content import ChatMessageContent
+from semantic_kernel.contents.utils.author_role import AuthorRole
+from semantic_kernel.functions.kernel_function_from_prompt import KernelFunctionFromPrompt
+from semantic_kernel.kernel import Kernel
+
+REVIEWER_NAME = "ArtDirector"
+REVIEWER_INSTRUCTIONS = """
+You are an art director who has opinions about copywriting born of a love for David Ogilvy.
+The goal is to determine if the given copy is acceptable to print.
+If so, state that it is approved.
+If not, provide insight on how to refine suggested copy without example.
+"""
+
+COPYWRITER_NAME = "CopyWriter"
+COPYWRITER_INSTRUCTIONS = """
+You are a copywriter with ten years of experience and are known for brevity and a dry humor.
+The goal is to refine and decide on the single best copy as an expert in the field.
+Only provide a single proposal per response.
+You're laser focused on the goal at hand.
+Don't waste time with chit chat.
+Consider suggestions when refining an idea.
+"""
+
+
+def _create_kernel_with_chat_completion(service_id: str) -> Kernel:
+    kernel = Kernel()
+    kernel.add_service(AzureChatCompletion(service_id=service_id))
+    return kernel
+
+
+async def main():
+    agent_reviewer = ChatCompletionAgent(
+        service_id="artdirector",
+        kernel=_create_kernel_with_chat_completion("artdirector"),
+        name=REVIEWER_NAME,
+        instructions=REVIEWER_INSTRUCTIONS,
+    )
+
+    agent_writer = ChatCompletionAgent(
+        service_id="copywriter",
+        kernel=_create_kernel_with_chat_completion("copywriter"),
+        name=COPYWRITER_NAME,
+        instructions=COPYWRITER_INSTRUCTIONS,
+    )
+
+    termination_function = KernelFunctionFromPrompt(
+        function_name="termination",
+        prompt="""
+        Determine if the copy has been approved.  If so, respond with a single word: yes
+
+        History:
+        {{$history}}
+        """,
+    )
+
+    selection_function = KernelFunctionFromPrompt(
+        function_name="selection",
+        prompt=f"""
+        Determine which participant takes the next turn in a conversation based on the the most recent participant.
+        State only the name of the participant to take the next turn.
+        No participant should take more than one turn in a row.
+        
+        Choose only from these participants:
+        - {REVIEWER_NAME}
+        - {COPYWRITER_NAME}
+        
+        Always follow these rules when selecting the next participant:
```

Comment:
I think there could be some improvements we can make here, as Ben suggested -- instead of parsing string output, we could look at a specific function returned to know when to terminate. Will discuss with Chris for a future improvement.